betty/app/__init__.py

```diff
@@ -35,30 +35,60 @@
     from concurrent.futures import Executor
     from betty.plugin import PluginRepository
     from betty.cache import Cache
-    from collections.abc import AsyncIterator, Callable
+    from collections.abc import AsyncIterator
 
 _T = TypeVar("_T")
 
 
+class _ShutdownHttpClient:
+    def __init__(self, http_client: aiohttp.ClientSession):
+        self._http_client = http_client
+
+    async def __call__(self, wait: bool):
+        await self._http_client.close()
+
+
 @final
 class App(Configurable[AppConfiguration], TargetFactory, ServiceProvider):
     """
     The Betty application.
     """
 
+    _binary_file_cache: BinaryFileCache
+    _cache: Cache[Any]
+
     def __init__(
         self,
         configuration: AppConfiguration,
         cache_directory_path: Path,
         *,
-        cache_factory: Callable[[Self], Cache[Any]],
+        cache_factory: ServiceFactory[Self, Cache[Any]],
         fetcher: Fetcher | None = None,
     ):
         super().__init__(configuration=configuration)
-        self._fetcher = fetcher
+        type(self).fetcher.init(self, fetcher)
         self._cache_directory_path = cache_directory_path
         self._cache_factory = cache_factory
 
+    def __getstate__(self) -> dict[str, Any]:
+        return {
+            "_binary_file_cache": getattr(self, "_binary_file_cache", None),
+            "_bootstrapped": self._bootstrapped,
+            "_cache": getattr(self, "_cache", None),
+            "_cache_directory_path": self._cache_directory_path,
+            "_cache_factory": self._cache_factory,
+            "_configuration": self._configuration,
+            # @todo Use a term like "computed" or "lazy" or what not? Because "from init" can mean different things again
```

Comment:
"implicit" versus "explicit"?

Reply:
Also think of factories provided through initializers (e.g. `App`'s cache factory).